app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -2,25 +2,36 @@ class ManageIQ::Providers::Vmware::InfraManager::Inventory::Collector
   include PropertyCollector
   include Vmdb::Logging
 
-  def initialize(ems, run_once: false)
+  attr_reader   :ems, :inventory_cache, :run_once, :saver
+  attr_accessor :exit_requested, :parser, :persister
+
+  def initialize(ems, run_once: false, threaded: true)
     @ems             = ems
     @inventory_cache = ems.class::Inventory::Cache.new
     @run_once        = run_once
+    @saver           = ems.class::Inventory::Saver.new(:threaded => threaded)
 
     self.exit_requested = false
+    self.parser         = nil
+    self.persister      = ems.class::Inventory::Persister.new(ems)
```

Comment:
a) parser, persister now have 3 different configurations. parser = nil, persister = value. and the setup below that sets both to a value. Then the end sets both to nil
b) too bad their life cycle has also changed.
c) concepts of saver vs persister is little confusing. But that is probably because naming is hard.

Reply:
maybe add here:

```
# Use the full refresh persister for the initial UpdateSet from WaitForUpdates	
# After the initial UpdateSet this will change to a targeted persister
```